tools/socktap/cohda.cpp

```diff
@@ -1,41 +1,149 @@
 #include "cohda.hpp"
-#include <vanetza/dcc/data_request.hpp>
-#include <cassert>
-#include <mk2mac-api-types.h>
+#include <boost/crc.hpp>
+#include <iostream>
+#include <llc-api.h>
+#include <vanetza/geonet/router.hpp>
+#include <vanetza/geonet/address.hpp>
+#include <vanetza/dcc/profile.hpp>
+#include <vanetza/common/byte_order.hpp>
 
-namespace vanetza
-{
+namespace vanetza {
 
-using dcc::Profile;
+struct LLCHeader {
+    uint8_t DSAP = 0xAA;
+    uint8_t SSAP = 0xAA;
+    uint8_t ControlField = 0x03;
+    uint8_t OUI[3] = { 0x00, 0x00, 0x00 };
+    uint16_t ProtocolID = geonet::ether_type.net();
+};
 
-ByteBuffer create_cohda_tx_header(const dcc::DataRequest& request)
-{
-    ByteBuffer header(sizeof(tMK2TxDescriptor), 0x00);
-    assert(header.size() == sizeof(tMK2TxDescriptor));
-    tMK2TxDescriptor* tx = reinterpret_cast<tMK2TxDescriptor*>(header.data());
-    tx->ChannelNumber = 180;
+/* See ETSI EN 302 663 V1.2.1 (2013-07), Table B.3 */
+enum tPriority {
+    AC_BK = 1, AC_BE = 3, AC_VI = 5, AC_VO = 6
```

Comment:
This table is a good hint. However, AC_BE is 0 then.
I am also thinking about changing `net::AccessCategory`...

Reply:
Since 802.11 allows values 0 and 3 for AC_BE, I would prefer 3, because it's the larger value and larger than AC_BK values, so a total order is maintained. Consequently, AC_VO should be changed to 7 for the same reasons that it's the larger of both possible values.

Reply:
Hm, maintaining a total order is a good point. I thought it would be nice then to keep a 1:1 relation of the names with 802.1d. Anyway, the numbering is odd and should not bother us too much in the end.

Reply:
So which one do you prefer:
- 1, 0, 5, 6 with a comment referencing the table and pointing out compatability to 802.1D, or
- 1, 3, 5, 7 with a commend referencing the table and pointing out the BE/BK resolution in 802.11?

Reply:
1, 3, 5, 7 looks good